### ISSUE TITLE
Cannot load Analyzers programatically from MSBuildWorkspace

### ISSUE THREAD

**Issue by shaggygi** [dotnet/roslyn#31797]
**Version Used**: 
VS 2017 (15.9.4)

Scenario:

Create a new .NET command-line project.
Add latest Microsoft.Build.Locator Nuget to project.
Add latest Microsoft.CodeAnalysis and StyleCop.Analyzers Nuget to project.
Project now has 4 Analyzer references:

```
Microsoft.CodeAnalysis.Analyzers
Microsoft.CodeAnalysis.CSharp.Analyzers
StyleCop.Analyzers
StyleCop.Analyzers.CodeFixes

// Add the following code to the project and run it:
MSBuildLocator.RegisterDefaults();
var workspace = MSBuildWorkspace.Create();
var project = await workspace.OpenProjectAsync(projectPath);

var allLanguageAnalyzers = project.AnalyzerReferences.SelectMany(r => r.GetAnalyzersForAllLanguages());
var languageSpecificAnalyzers = project.AnalyzerReferences.SelectMany(r => r.GetAnalyzers(project.Language));
```

Where projectPath is the path to this project.

Expectations:

allLanguageAnalyzers is not empty
languageSpecificAnalyzers is not empty
Actual Results:

allLanguageAnalyzers is empty
languageSpecificAnalyzers is empty

This might be similar to [this older issue](https://github.com/dotnet/roslyn/issues/7654), but not sure, so I'm pinging the following for help.

@yaakov-h
@jmarolf
@davkean 
@sharwell

**Comment by jmarolf** [comment]
@shaggygi 	what is `projectPath` pointing to?  Also, that is the XML for you project file.

**Comment by shaggygi** [comment]
@jmarolf sorry for not showing that.  It is a path to a .csproj file. I’ve tried console and wpf projects with same result.

**Comment by jmarolf** [comment]
@shaggygi and the project pointed to in `projectPath` has which analyzer references?

**Comment by shaggygi** [comment]
@jmarolf The app that runs the code includes Microsoft.CodeAnalysis and StyleCop.Analyzers and the project the code is check only has StyleCop.Analyzers added.

**Comment by shaggygi** [comment]
@jmarolf I just publish a sample here if this helps.  https://github.com/shaggygi/TryingAnalyzerReferences

**Comment by shaggygi** [comment]
Not sure if this is related.  I noticed when I do thing following emit, the only diagnostics returned are with Hidden severity that point to things like *.g.* files.  It seems like you would also get diagnostics that come from StyleCop.Analyzers.

`EmitResult emitResult = compilation.Emit(stream);`

**Comment by JPCMarques** [comment]
Any updates on this?

**Comment by jasonmalinowski** [comment]
Analyzers should be loading; this was probably some sort of binding redirect issue or something else specific to that setup.

**Comment by filzrev** [comment]
Unresolved analyzers are represented as `UnresolvedAnalyzerReference`.
And GetAnalyzers/GetAnalyzersForAllLanguages returns empty results.

So, if it need to detect unresolved analyzers.
It need manuall handle `UnresolvedAnalyzerReference`.

```csharp
foreach (var unresolvedAnalyzer in project.AnalyzerReferences.OfType<UnresolvedAnalyzerReference>())
{
    Console.WriteLine($"Failed to resolve analyzer:  {unresolvedAnalyzer.FullPath}");
}
```

**Comment by jasonmalinowski** [comment]
@filzrev I wouldn't expect unresolved analyzers to be happening unless we ended up with a file path that doesn't exist. If you're seeing that then maybe we have a new bug (and please file one!)

**Comment by filzrev** [comment]
>  I wouldn't expect unresolved analyzers to be happening unless we ended up with a file path that doesn't exist. 

Yes, that's right, it happens when the analyzer DLL is not found.

On some use cases. analyzer silently failed to load. (https://github.com/dotnet/docfx/issues/9619 and #77255)
So It need to manually handle these cases manually.

- https://github.com/dotnet/docfx/issues/9619
- https://github.com/dotnet/docfx/blob/04746a10c565dd2566c8ee064fe23026022eed01/src/Docfx.Dotnet/DotnetApiCatalog.Compile.cs#L146-L164

**Comment by jasonmalinowski** [comment]
@filzrev Ah yes, that's another good case to capture in #77980.
